### PR TITLE
feat(auth-server): add route for getCheckout Token

### PR DESCRIPTION
**Pull request by birdsarah**
## Because

- Route must be added to get paypal checkout token

## This pull request

- Adds the route

## Issue that this pull request solves

Closes: # (issue number)

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/fxa-auth-server/lib/routes/subscriptions/paypal.ts [hunk 1]
```diff
@@ -26,6 +27,13 @@ export class PayPalHandler extends StripeHandler {
     super(log, db, config, customs, push, mailer, profile, stripeHelper);
     this.paypalHelper = Container.get(PayPalHelper);
   }
+
+  /**
+   * Get PayPal Checkout Token.
+   */
+  async getCheckoutToken() {
+    return await this.paypalHelper.getCheckoutToken();
```
Comment (bbangert) [inline packages/fxa-auth-server/lib/routes/subscriptions/paypal.ts:35]: There's a few calls that are pretty standard on every http call:
https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts#L178

The minimum is generally the log statement, and the customs check (prevents abuse of our endpoints).

File: packages/fxa-auth-server/lib/routes/subscriptions/paypal.ts [hunk 2]
```diff
@@ -49,8 +57,20 @@ export const paypalRoutes = (
...
+    {
+      method: 'POST',
+      path: '/oauth/subscriptions/paypal-checkout',
+      options: {
+        auth: {
+          payload: false,
+          strategy: 'oauthToken',
+        },
+        response: {
+          schema: isA.string(),
```
Comment (bbangert) [inline packages/fxa-auth-server/lib/routes/subscriptions/paypal.ts:70]: Our endpoints return JSON, so I'd expect the schema here to be an object with a token property.

**Review by bbangert — CHANGES_REQUESTED** [review]
Two minor changes, other than that this looks great!

File: packages/fxa-auth-server/lib/routes/subscriptions/paypal.ts [hunk 3]
```diff
@@ -49,8 +64,23 @@ export const paypalRoutes = (
...
-  //        overall request deadline. Stripe retries must include a idempotency_key.
-  return [];
+  return [
+    {
+      method: 'POST',
+      path: '/oauth/subscriptions/paypal-checkout',
+      options: {
+        auth: {
+          payload: false,
+          strategy: 'oauthToken',
```
Comment (birdsarah) [inline packages/fxa-auth-server/lib/routes/subscriptions/paypal.ts:74]: Is this is the correct strategy? I don't know which to pick and I'm not sure how to write a test for it.

Comment (bbangert) [reply packages/fxa-auth-server/lib/routes/subscriptions/paypal.ts:74]: No test is needed for this bit.

**Comment by bbangert** [comment]
The CI errors look legitimate for the TypeScript issues.

**Review by bbangert — APPROVED** [review]

**Pull request merged**